File: NetFramework/TreeView/VB/CustomEditors/Editors/EditorsData.vb
Imports System.ComponentModel

Public Class EditorsData

    Public Property TextValue As String
    Public Property ObjectValue As Object
    Public Property ColorValue As Color
    Public Property DateValue As DateTime
    Public Property Collection As BindingList(Of EditorsData)

    Public Sub New()
        Collection = New BindingList(Of EditorsData)
    End Sub

    Public Shared Function GetData() As BindingList(Of EditorsData)
        Dim Data = New BindingList(Of EditorsData)
        For i As Integer = 0 To 5
            Dim ed = New EditorsData()
            ed.TextValue = i.ToString()
            ed.ObjectValue = i
            ed.ColorValue = Color.Black
            ed.DateValue = DateTime.Now
            Data.Add(ed)
            For j As Integer = 0 To 5
                Dim ed1 = New EditorsData()
                ed1.TextValue = (i + j).ToString()
                ed1.ObjectValue = i + j
                ed1.ColorValue = Color.White
                ed1.DateValue = DateTime.Now
                Data(i).Collection.Add(ed1)
            Next
        Next
        Return Data
    End Function

End Class

File: NetFramework/TreeView/CS/CustomEditors/Editors/EditorsData.cs
using System;
using System.ComponentModel;
using System.Drawing;

namespace SamplesData.Editors
{
    public class EditorsData
    {
        public string TextValue
        {
            get; set;
        }

        public object ObjectValue
        {
            get; set;
        }

        public Color ColorValue
        {
            get; set;
        }

        public DateTime DateValue
        {
            get; set;
        }

        public BindingList<EditorsData> Collection
        { get; set; }

        public EditorsData()
        {
            Collection = new BindingList<EditorsData>();
        }

        public static BindingList<EditorsData> GetData()
        {
            var data = new BindingList<EditorsData>();
            for (int i = 0; i < 5; i++)
            {
                data.Add(new EditorsData() { TextValue = i.ToString(), ObjectValue = i, ColorValue = Color.Black, DateValue = DateTime.Now });
                for (int j = 0; j < 5; j++)
                {
                    data[i].Collection.Add(new EditorsData() { TextValue = (i + j).ToString(), ObjectValue = i + j, ColorValue = Color.White, DateValue = DateTime.Now });
                }
            }

            return data;
        }
    }
}
